src/rules/whitespaceRule.ts

```diff
@@ -122,6 +128,21 @@ function walk(ctx: Lint.WalkContext<Options>) {
                 if (options.preblock) {
                     checkForTrailingWhitespace(node.getFullStart());
                 }
+                if (options.postblockPrekeyword) {
+                    const nextToken = utils.getNextToken(node as ts.Block)!;
+                    if (!utils.isSameLine(sourceFile, node.end, nextToken.getStart())) {
+                        break;
+                    }
+                    switch (nextToken.kind) {
+                        case ts.SyntaxKind.WhileKeyword:
```

Comment:
This may do a bit too much. That `WhileKeyword` could be from the `while (...) {}` statement following the block.
That may not be a big deal, because it will only complain about such code:
```ts
if (condition) { /* code */ }while(condition) {};
                             ~ [missing whitespace]
```

Anyway, there's an easier way to do this check without `getNextToken`:

```ts
const parent = node.parent!;
if (isDoStatement(parent) || // Block is followed by WhiteKeyword
    isIfStatement(parent) && parent.elseStatement !== undefined && parent.thenStatement === node || // Block is followed by ElseKeyword
    isTryStatement(parent) && (
        parent.tryBlock === node || // Block is followed by CatchKeyword or FinallyKeyword
        parent.catchClause !== undefined && parent.catchClause.block === node // Block is followed by FinallyKeyword
    )
) {
    // do stuff
}
```
Well, while writing it down I realized it's not really easier. But it's clearer what kind of statement is checked.

Reply:
OK. I wrote a function to check conditions as you mentions.